### PR TITLE
SI-7143 Fix scanner docComment production.

### PR DESCRIPTION
**Pull request by vigdorchik**
Wasn't able to extract a test case from the failing IDE tests unfortunately.
Review by @dragos.

### CODE REVIEW

**Comment by vigdorchik** [comment]
To duplicate from the request: the NPE I'm after is

java.lang.NullPointerException
at scala.tools.nsc.ast.parser.Parsers$Parser$$anonfun$2.apply(Parsers.scala:668)
at scala.tools.nsc.ast.parser.Parsers$Parser$$anonfun$2.apply(Parsers.scala:665)
at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
at scala.collection.immutable.List.foreach(List.scala:318)
at scala.collection.TraversableLike$class.map(TraversableLike.scala:244)
at scala.collection.AbstractTraversable.map(Traversable.scala:105)
at scala.tools.nsc.ast.parser.Parsers$Parser.joinComment(Parsers.scala:664)
at scala.tools.nsc.ast.parser.Parsers$Parser.topStatSeq(Parsers.scala:2937)
at scala.tools.nsc.ast.parser.Parsers$Parser$$anonfun$compilationUnit$1.topstats$1(Parsers.scala:3132)
at scala.tools.nsc.ast.parser.Parsers$Parser$$anonfun$compilationUnit$1.topstats$1(Parsers.scala:3124)
at scala.tools.nsc.ast.parser.Parsers$Parser$$anonfun$compilationUnit$1.apply(Parsers.scala:3138)
at scala.tools.nsc.ast.parser.Parsers$Parser$$anonfun$compilationUnit$1.apply(Parsers.scala:3102)
at scala.tools.nsc.ast.parser.Parsers$Parser.checkNoEscapingPlaceholders(Parsers.scala:428)
at scala.tools.nsc.ast.parser.Parsers$Parser.compilationUnit(Parsers.scala:3102)
at scala.tools.nsc.ast.parser.Parsers$SourceFileParser$$anonfun$parseStartRule$1.apply(Parsers.scala:141)
at scala.tools.nsc.ast.parser.Parsers$SourceFileParser$$anonfun$parseStartRule$1.apply(Parsers.scala:141)
at scala.tools.nsc.ast.parser.Parsers$Parser.parse(Parsers.scala:313)
at scala.tools.nsc.symtab.BrowsingLoaders.browseTopLevel(BrowsingLoaders.scala:104)
at scala.tools.nsc.symtab.BrowsingLoaders.enterToplevelsFromSource(BrowsingLoaders.scala:119)
at scala.tools.nsc.symtab.SymbolLoaders.initializeFromClassPath(SymbolLoaders.scala:136)
at scala.tools.nsc.symtab.SymbolLoaders$PackageLoader$$anonfun$doComplete$3.apply(SymbolLoaders.scala:232)
at scala.tools.nsc.symtab.SymbolLoaders$PackageLoader$$anonfun$doComplete$3.apply(SymbolLoaders.scala:231)
at scala.collection.TraversableLike$WithFilter$$anonfun$foreach$1.apply(TraversableLike.scala:772)
at scala.collection.Iterator$class.foreach(Iterator.scala:727)
at scala.collection.AbstractIterator.foreach(Iterator.scala:1157)
at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:771)
at scala.tools.nsc.symtab.SymbolLoaders$PackageLoader.doComplete(SymbolLoaders.scala:231)
at scala.tools.nsc.symtab.SymbolLoaders$SymbolLoader.complete(SymbolLoaders.scala:187)
at scala.reflect.internal.Symbols$Symbol.info(Symbols.scala:1229)
at scala.reflect.internal.Types$TypeRef.decls(Types.scala:2428)
at scala.tools.nsc.typechecker.Namers$Namer.enterPackage(Namers.scala:630)
at scala.tools.nsc.typechecker.Namers$Namer.dispatch$1(Namers.scala:233)
at scala.tools.nsc.typechecker.Namers$Namer.liftedTree1$1(Namers.scala:248)
at scala.tools.nsc.typechecker.Namers$Namer.enterSym(Namers.scala:248)
at scala.tools.nsc.typechecker.Analyzer$namerFactory$$anon$1.apply(Analyzer.scala:43)
at scala.tools.nsc.Global$GlobalPhase.applyPhase(Global.scala:463)
at scala.tools.nsc.Global$Run$$anonfun$compileLate$2$$anonfun$apply$1.apply$mcV$sp(Global.scala:1682)
at scala.tools.nsc.Global$Run$$anonfun$compileLate$2$$anonfun$apply$1.apply(Global.scala:1682)
at scala.tools.nsc.Global$Run$$anonfun$compileLate$2$$anonfun$apply$1.apply(Global.scala:1682)
at scala.reflect.internal.SymbolTable.atPhase(SymbolTable.scala:208)
at scala.tools.nsc.Global$Run$$anonfun$compileLate$2.apply(Global.scala:1682)
at scala.tools.nsc.Global$Run$$anonfun$compileLate$2.apply(Global.scala:1681)
at scala.collection.Iterator$class.foreach(Iterator.scala:727)
at scala.collection.AbstractIterator.foreach(Iterator.scala:1157)
at scala.tools.nsc.Global$Run.compileLate(Global.scala:1681)
at scala.tools.nsc.interactive.Global.scala$tools$nsc$interactive$Global$$parseAndEnter(Global.scala:573)
at scala.tools.nsc.interactive.Global$$anonfun$backgroundCompile$3$$anonfun$apply$6.apply(Global.scala:483)
at scala.tools.nsc.interactive.Global$$anonfun$backgroundCompile$3$$anonfun$apply$6.apply(Global.scala:480)
at scala.Option.foreach(Option.scala:236)
at scala.tools.nsc.interactive.Global$$anonfun$backgroundCompile$3.apply(Global.scala:480)
at scala.tools.nsc.interactive.Global$$anonfun$backgroundCompile$3.apply(Global.scala:480)
at scala.collection.immutable.List.foreach(List.scala:318)
at scala.tools.nsc.interactive.Global.backgroundCompile(Global.scala:480)
at scala.tools.nsc.interactive.PresentationCompilerThread.run(PresentationCompilerThread.scala:25)

**Comment by dragos** [comment]
It looks good, but the comment message is not very clear. Could you amend it explaining what was the failing scenario? Sometimes it's useful to know why a certain change was necessary.

File: src/compiler/scala/tools/nsc/ast/parser/Scanners.scala [hunk 1]
```diff
@@ -554,24 +546,23 @@ trait Scanners extends ScannersCommon {
           var buildingDocComment = false
           if (ch == '*' && buildDocs) {
             buildingDocComment = true
-            docBuffer = new StringBuilder("/**")
```
Comment (huitseeker) [inline src/compiler/scala/tools/nsc/ast/parser/Scanners.scala:557]: I missed how you don't lose that leading `/**` in the final output of `flushDoc`, now ? What have I overlooked ?

Comment (vigdorchik) [inline src/compiler/scala/tools/nsc/ast/parser/Scanners.scala:557]: foundDocComment contains the full text of the comment, including leading "/*".

Comment (huitseeker) [inline src/compiler/scala/tools/nsc/ast/parser/Scanners.scala:557]: Got it. For future me : `skipComment` is triggered after seeing `/`, this `if` catches the _second_ star (third character) of the comment, and by then, `comment` contains `/**`.

**Comment by huitseeker** [comment]
LGTM!

**Pull request merged**